### PR TITLE
omelasticsearch: fix bulk maxbytes boundaries

### PR DESCRIPTION
**Pull request by rgerhards**
## Summary

- Account for all conditional bulk metadata when enforcing `maxbytes`.
- Avoid an empty bulk POST when the first record exceeds `maxbytes`.
- Add a fake-HTTP regression test for both boundary paths.

## Validation

- Host bootstrap check: `make -j10 check TESTS=""`.
- Regression test: `make -C tests check TESTS=omelasticsearch-maxbytes-boundary.sh`.
- Ubuntu 26.04 change-gated container CI: 1412 passed, 74 skipped, 0 failed, 0 errors.
- C format, shell syntax, ShellCheck, and checkbashisms passed.
- Static analyzer reports the known unrelated `runtime/modules.c:412` baseline finding; its standard lane disables omelasticsearch.

<a href="https://cubic.dev/pr/rsyslog/rsyslog/pull/7542?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: tests/omelasticsearch-maxbytes-boundary.sh [hunk 1]
```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+# Regression test for omelasticsearch bulk maxbytes boundaries. The local HTTP
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply tests/omelasticsearch-maxbytes-boundary.sh:6]: Done in 988e077f1.

File: tests/omelasticsearch-maxbytes-boundary.sh [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+
+server = HTTPServer(("127.0.0.1", 0), Handler)
+with open(port_file, "w", encoding="ascii") as fh:
+    fh.write(f"{server.server_port}\n")
+server.serve_forever()
+PY
+SERVER_PID=$!
+
+wait_file_exists "$PORT_FILE" 10
```
[the remark this answers is not in the document]
Comment (rgerhards) [reply tests/omelasticsearch-maxbytes-boundary.sh]: Done in 988e077f1; the second argument is now an accurate timeout diagnostic.
